### PR TITLE
client: send request source info only when clone source file segment is allocated and file status is CloneMetaInstalled

### PR DESCRIPTION
**Pull request by wu-hanqing**
### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?

What's Changed:

Previously, when client reads and writes data within the length of the source file, rpc will carry the source file information.
Now, only when client reads and writes data within the range that source file' segment is allocated, and file status is CloneMetaInstalled, rpc will carry the source file information.

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/mds/nameserver2/namespace_service.cpp [hunk 1]
```diff
@@ -1431,30 +1439,30 @@ void NameSpaceService::RefreshSession(
...
-            slash = true;
-        } else {
-            slash = false;
-        }
-    }
-
-    // if some other limits to path can add here in the future
-    return true;
-}
+// bool isPathValid(const std::string path) {
```
Comment (ilixiaocui) [inline src/mds/nameserver2/namespace_service.cpp:1442]: 不要的就删掉吧

File: src/client/client_common.h [hunk 2]
```diff
@@ -281,6 +310,52 @@ class ClientDummyServerInfo {
...
+    if (location.IsValid()) {
+        os << location.cloneFileSource << ":" << location.cloneFileOffset;
+    } else {
+        os << "empty";
+    }
+
+    return os;
+}
+
+inline RequestSourceInfo CloneSourceInfo::GetRequestSourceInfo(
```
[the remark this answers is not in the document]
Comment (wu-hanqing) [reply src/client/client_common.h]: Done

File: src/client/lease_executor.cpp [hunk 3]
```diff
@@ -40,6 +40,8 @@ LeaseExecutor::LeaseExecutor(const LeaseOption& leaseOpt,
     iomanager_   = iomanager;
     leaseoption_ = leaseOpt;
     task_ = nullptr;
+
+    metaCache_ = iomanager_->GetMetaCache();
```
[the remark this answers is not in the document]
Comment (wu-hanqing) [reply src/client/lease_executor.cpp]: Done

File: src/client/lease_executor.h [hunk 4]
```diff
@@ -125,10 +125,11 @@ class LeaseExecutor {
     void IncremRefreshFailed();
 
     /**
-     * 每次续约会携带新的文件信息，该函数检查当前文件是否需要更新版本信息
-     * @param: newversion是lease续约后从mds端携带回来的版本号
+     * @brief 每次续约会携带新的文件信息，更新本地文件信息与mds记录一致，
+     *        当前只更新sn以及status
```
[the remark this answers is not in the document]
Comment (wu-hanqing) [reply src/client/lease_executor.h]: Done

File: src/client/splitor.cpp [hunk 5]
```diff
@@ -297,23 +297,19 @@ bool Splitor::GetOrAllocateSegment(bool allocateIfNotExist,
...
 
-    uint64_t offset = static_cast<uint64_t>(chunkIdx) * fileInfo->chunksize;
-
-    if (offset >= fileInfo->cloneLength) {
-        return {};
+    const FInfo* fileInfo = metaCache->GetFileInfo();
+    if (fileInfo->filestatus == FileStatus::CloneMetaInstalled) {
+        const CloneSourceInfo& sourceInfo = fileInfo->sourceInfo;
+        uint64_t offset = static_cast<uint64_t>(chunkIdx) * fileInfo->chunksize;
+        return sourceInfo.GetRequestSourceInfo(offset);
```
[the remark this answers is not in the document]
Comment (wu-hanqing) [reply src/client/splitor.cpp]: Done

File: src/mds/nameserver2/namespace_service.cpp (hunk 1 above)
Comment (wu-hanqing) [reply src/mds/nameserver2/namespace_service.cpp]: Done

**Review by ilixiaocui — APPROVED** [review]

File: src/mds/nameserver2/curvefs.cpp [hunk 6]
```diff
@@ -1370,6 +1374,70 @@ StatusCode CurveFS::OpenFile(const std::string &fileName,
 
     fileRecordManager_->GetRecordParam(protoSession);
 
+    // clone file
+    if (fileInfo->has_clonesource()) {
```
Comment (xu-chaojie) [inline src/mds/nameserver2/curvefs.cpp:1378]: 可以封装个函数，这一段太长了

**Comment by wu-hanqing** [comment]
recheck

**Pull request merged**